### ISSUE TITLE
Suppport for adapterOptions on modelActions

### ISSUE THREAD

**Issue by scottwernervt** [Exelord/ember-custom-actions#24]
I use a dedicated endpoint  `/api/me` for managing the current user as setup in [ember-simple-auth/guides/managing-current-user](https://github.com/simplabs/ember-simple-auth/blob/master/guides/managing-current-user.md). Updating attributes on the current user model is done by: `currentUser.save({ adapterOptions: { me: true } })`. Overriding the `urlFor*` methods on `ApplicationAdapter` replaces the `id` with `me` before making the request.

Is it possible to add support for `adapterOptions` when calling a `modelAction`? For example: 
```js
const me = get(this, 'currentUser.user');
me.unreadNotificationCount({ adapterOptions: { me: true } }).then((response) => ...);
```

As a workaround, I create an empty user model with `id = me` (similiar to example for _Resource actions_):

```js
const me = get(this, 'store').createRecord('user', { id: 'me' });
me.unreadNotificationCount({ adapterOptions: { me: true } }).then((response) => ...);
```

**Comment by Exelord** [comment]
Hmm, that's interesting.
So, simply you want to make a request to `api/users/me` and want to use a custom `id` or a request.

I can consider adding that option but for now, you can use `resourceAction` as a much simpler and meaningful way.
Example

```js
// model
unreadNotificationCount: resourceAction('me')

// use
let currentUser = this.get('currentUser.user');
currentUser.unreadNotificationCount(); // request to -> 'api/users/me'
```

Is that the expected behavior? Correct me if I'm wrong :)

Btw,
Is `adapterOptions` documented somewhere in ember data?

**Comment by scottwernervt** [comment]
> So, simply you want to make a request to api/users/me and want to use a custom id or a request.

Yes, that would be a case for adapterOptions. I would like for ember-custom-actions to pass `adapterOptions: {}` when performing a `resourceAction` or `modelAction`. For example, if the user wanted to include query parameters or override any logic in the `ApplicationAdapter` class.

> Is adapterOptions documented somewhere in ember data?

The documentation for it is sparse. Here are some resources:

- [Add a way to pass options to an adapter](https://github.com/emberjs/data/pull/3278)
- [DS.JSONAPIAdapter.urlForFindRecord(id, modelName, snapshot)](https://emberjs.com/api/ember-data/2.14/classes/DS.JSONAPIAdapter/methods/urlForFindRecord?anchor=urlForFindRecord) for an example of `adapterOptions` being included on the `snapshot` argument.
- [model.save(options)](https://github.com/emberjs/data/blob/master/addon/-private/system/model/model.js)
- [Ember Data: Passing query params to .save()](http://blog.shipshape.io/ember-data-passing-query-params-to-save-2/)

**Comment by Exelord** [comment]
Ok, we can add `adapterOptions` to `buildUrl` function but...
Adapters were not designed to work with custom actions.That's why the current implementation use adapters only for the resource base path and not rely on them.

- `api/users` will be taken from adapter even if we pass the options because we can not define custom `urlFor` for custom actions, for now.
- you can always pass `queryParams` using the `params` key in options. more info https://github.com/Exelord/ember-custom-actions#params

Could you share with me your exact case what are you trying to archive? Right now, there is everything possible except the custom base path (which is very easy to implement)

**Comment by Exelord** [comment]
OK I think your case may be covered in #25 when it will be implemented

**Comment by Exelord** [comment]
Implemented in #28.
